### PR TITLE
Add `ignore-self-only-init` option

### PR DESCRIPTION
**Pull request by thejcannon**
Adding an option `ignore-self-only-init` (off by default) which will not report a D107 error if `__init__` has no parameters other than `self`. The justification is that these methods usually aren't very interesting and will likely result in redundant/boring docstrings which add noise.

Just turning off D107 isn't an option, since there are `__init__` methods which _are_ interesting ;)

Please make sure to check for the following items:
- [x] Add unit tests and integration tests where applicable.  
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".

### CODE REVIEW

**Comment by thejcannon** [comment]
Of course, the option name is up for debate :)

File: src/pydocstyle/parser.py [hunk 1]
```diff
@@ -665,8 +671,10 @@ def parse_definition(self, class_):
         name = self.current.value
         self.log.debug("parsing %s '%s'", class_.__name__, name)
         self.stream.move()
+        call_names = []
```
Comment (samj1912) [inline src/pydocstyle/parser.py:674]: Should we name this `callable_args` instead?

**Review by samj1912 — COMMENTED** [review]
Hey @thejcannon - thanks for the PR! Sorry for the delay in getting to review this. It generally looks good to me. Just a minor comment re:naming.

File: src/pydocstyle/parser.py (hunk 1 above)
Comment (thejcannon) [reply src/pydocstyle/parser.py:674]: At this point I forgot why I called it this :joy:

`callable_args` it is!

**Comment by samj1912** [comment]
@thejcannon could you please rebase this PR against the master branch? Thanks!

**Comment by thejcannon** [comment]
@samj1912 got time for a quick review?

File: docs/release_notes.rst [hunk 2]
```diff
@@ -35,6 +35,7 @@ New Features
 * Allow for hanging indent when documenting args in Google style. (#449)
 * Add support for `property_decorators` config to ignore D401.
 * Add support for Python 3.10 (#554).
+* Add `ignore-self-only-init` config (#560).
```
Comment (samj1912) [inline docs/release_notes.rst:38]: Can we move this to the appropriate version section?

**Review by samj1912 — APPROVED** [review]
Minor comment about release notes. Lgtm apart from that.

File: docs/release_notes.rst (hunk 2 above)
Comment (thejcannon) [reply docs/release_notes.rst]: done

**Pull request merged**